app/models/poll.rb

```diff
@@ -10,12 +12,43 @@ class Poll < ActiveRecord::Base
 
   has_many :options, dependent: :destroy
 
+  serialize :vote_results, Array
+  serialize :current_state, Array
+
   validates :title, presence: true
 
   after_touch :ensure_status_is_correct
 
   def ready!
-    options.empty? ? errors.add(:status, "can't be ready when poll have no options") : super
+    if options.empty?
+      errors.add(:status, "can't be ready when poll have no options")
+    else
+      self.vote_results = []
+      self.current_state = []
+      super
+    end
+  end
+
+  def draft!
+    self.vote_results = []
+    self.current_state = []
+    super
+  end
+
+  def vote!
+    vote = SchulzeBasic.do vote_results, vote_results.first.count
+    self.current_state = vote.ranks
+    save
+  end
+
+  def save_preferences_as_weight(arr)
+    arr = arr.map { |opt| opt.split('_').last.to_i }.reverse
```

Comment:
This is clearly some external data format parsing and sanitation, and should be placed in controller logic.